runtime/src/main/java/io/quarkiverse/cxf/CxfClientProducer.java

```diff
@@ -27,26 +33,46 @@
 import org.apache.cxf.wsdl.ExtensionClassLoader;
 import org.jboss.logging.Logger;
 
-@ApplicationScoped
-public class CxfClientProducer {
+import io.quarkiverse.cxf.annotation.CXFClient;
 
+/**
+ * Base producer class for setting up CXF client proxies.
+ * <p>
+ * During augementation (build-time) a bean is created derived from this class for each SEI. The producing method calls
+ * loadCxfClient() to get a WS client proxy.
+ * <p>
+ * Notice the InjectionPoint parameter present in signature of loadCxfClient. Via that meta information we calculate the
+ * proper configuration to use.
+ */
+public abstract class CxfClientProducer {
     private static final Logger LOGGER = Logger.getLogger(CxfClientProducer.class);
+    private static final String DEFAULT_EP_ADDR = "http://localhost:8080";
 
-    public CXFClientInfo getInfo() {
-        return null;
-    }
+    @Inject
+    CxfConfig config;
 
-    public Object loadCxfClient() {
+    /**
+     * Must be public, otherwise: java.lang.VerifyError: Bad access to protected data in invokevirtual
+     */
+    public Object loadCxfClient(InjectionPoint ip, CXFClientInfo meta) {
+        return produceCxfClient(selectorCXFClientInfo(config, ip, meta));
+    }
 
-        CXFClientInfo cxfClientInfo = getInfo();
+    /**
+     * The main workhorse producing a CXF client proxy.
+     *
+     * @param info
+     * @return
+     */
+    private Object produceCxfClient(CXFClientInfo info) {
```

Comment:
Kindly change `CXFClientInfo info` to `CXFClientInfo cxfClientInfo` as it improves readability and maintainability, and it's also consistent with existing standards in the code base.

Reply:
Fixed.

Reply:
Done.